src/metabase/automagic_dashboards/core.clj

```diff
@@ -1163,13 +1164,36 @@
                    (:full-name root))
                  (tru "where {0}" (humanize-filter-value root cell-query))]))
 
+(defn- key-in?
+  "Recursively finds key in coll, returns true or false"
+  [coll k]
+  (boolean (let [coll-zip (zip/zipper coll? #(if (map? %) (vals %) %) nil coll)]
+    (loop [x coll-zip]
+      (when-not (zip/end? x)
+        (if-let [v (k (zip/node x))] true (recur (zip/next x))))))))
```

Comment:
Can joins be hidden in lower levels? Or are they always at `[:query :joins]`?

Reply:
yes

https://github.com/metabase/metabase/blob/be98b105ac914bba6274049beff691ae9058defa/shared/src/metabase/mbql/schema.cljc#L1032

Reply:
Also, probably more standard to use the match utilies on mbql
```clojure
match=> (match-one {:database 234,
                    :type :query,
                    :query
                    {:fields [[:field 33845 nil] [:field 33847 {:join-alias "f"}]],
                     :joins
                     [{:source-table 10804,
                       :condition [:= [:field 33844 nil] [:field 33848 {:join-alias "f"}]],
                       :strategy :left-join,
                       :alias "f"}],
                     :order-by [[:asc [:field 33845 nil]]],
                     :source-table 10803}}
                   {:joins [joins]} joins)
{:source-table 10804, :condition [:= [:field 33844 nil] [:field 33848 {:join-alias "f"}]], :strategy :left-join, :alias "f"}
```

Alth

Reply:
oh i see my confusion. Why is `splice-in` conditioned on the presence of `join-alias`? If the root has a join, wouldn't we want all of the cards to include that join?

Reply:
there's other functionality to put in joins after the fact in pretty disparate portions of the xray codebase and I didn't want funky action-at-a-distance shenanigans